mapie/metrics.py

```diff
@@ -307,3 +311,441 @@ def top_label_ece(
         )
     ece /= len(labels)
     return ece
+
+
+def regression_coverage_score_v2(
+    y_true: NDArray,
+    y_intervals: NDArray,
+) -> NDArray:
+    """
+    Effective coverage score obtained by the prediction intervals.
+
+    The effective coverage is obtained by estimating the fraction
+    of true labels that lie within the prediction intervals.
+
+    It is different from ``regression_coverage_score`` because it uses
+    directly the output of ``predict`` method and can compute the
+    coverage for each alpha.
+
+    Parameters
+    ----------
+    y_true: NDArray of shape (n_samples, n_alpha)
+        True labels.
+    y_intervals: NDArray of shape (n_samples, 2, n_alpha)
+        Lower and upper bound of prediction intervals
+        with different alpha risks.
+
+    Returns
+    -------
+    NDArray of shape (n_alpha,)
+        Effective coverage obtained by the prediction intervals.
+
+    Raises
+    ------
+    ValueError
+        If y_true is not of shape (n_samples, n_alpha)
+    """
+    y_intervals = check_array_shape_regression(y_true, y_intervals)
```

Comment:
Why do we ask the user to give y_true of shape (n_samples, n_alphas) ? the repeat of y_true should be done within the function

Reply:
y_true does not have necessarily the same values for each alpha. For example, in regression_ssc, regression_coverage_score_v2 is called for each bin and the smallest intervals may not be the same for alpha1 and alpha2. In this case y_true[:, 0] and y_true[:, 1] are not the same.

Reply:
Do we talk about true targets or predicted targets?

Reply:
True targets

Reply:
I don't understand how the value of y_true can change with alpha

Reply:
It can now take y_true of shapes (n_samples, n_alphas) and (n_samples,)